exporter/coralogixexporter/client.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+
+	cxpb "github.com/coralogix/opentelemetry-cx-protobuf-api/coralogixpb"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// CoralogixClient by Coralogix
+type coralogixClient struct {
+	cfg      Config
+	logger   *zap.Logger
+	conn     *grpc.ClientConn
+	client   cxpb.CollectorServiceClient
+	metadata cxpb.Metadata
+}
+
+// NewCoralogixClient by Coralogix
+func newCoralogixClient(cfg *Config, logger *zap.Logger) *coralogixClient {
+	c := &coralogixClient{
+		cfg:    *cfg,
+		logger: logger,
+		metadata: cxpb.Metadata{
+			ApplicationName: cfg.AppName,
+			SubsystemName:   cfg.SubSystem,
+		},
+	}
+	c.startConnection()
+	return c
+}
+func (c *coralogixClient) startConnection() {
+	confTLS := &tls.Config{
+		InsecureSkipVerify: false,
+	}
+	conn, err := grpc.Dial(c.cfg.GRPCClientSettings.Endpoint, grpc.WithTransportCredentials(credentials.NewTLS(confTLS)))
```

Comment:
See other exporters how they build the gRPC connection. Basicall, you'd call the GRPCClientSettings.ToDialOptions().

Reply:
Thanks, I will change this method in the next commit.